internal/engine/synclet_build_and_deployer.go

```diff
@@ -160,31 +156,61 @@ func (sbd *SyncletBuildAndDeployer) PostProcessBuild(ctx context.Context, manife
 		logger.Get(ctx).Infof("can't get container for for '%s': BuildResult has no image", manifest.Name)
 		return
 	}
-	if _, ok := sbd.getContainerIDForManifest(manifest.Name); !ok {
-		cID, err := sbd.getContainerForBuild(ctx, result)
+	if _, ok := sbd.getDeployInfoForManifest(manifest.Name); !ok {
+		deployInfo, err := sbd.getDeployInfo(ctx, result.Image)
 		if err != nil {
-			logger.Get(ctx).Infof("couldn't get container for %s: %v", manifest.Name, err)
+			logger.Get(ctx).Infof("failed to get deployInfo: %v", err)
 			return
 		}
-		sbd.setContainerIDForManifest(manifest.Name, cID)
+		sbd.setDeployInfoForManifest(manifest.Name, deployInfo)
 	}
 }
 
-func (sbd *SyncletBuildAndDeployer) getContainerForBuild(ctx context.Context, build BuildResult) (k8s.ContainerID, error) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, "SyncletBuildAndDeployer-getContainerForBuild")
+func (sbd *SyncletBuildAndDeployer) getDeployInfo(ctx context.Context, image reference.NamedTagged) (DeployInfo, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "SyncletBuildAndDeployer-getDeployInfo")
 	defer span.Finish()
 
 	// get pod running the image we just deployed
-	pID, err := sbd.kCli.PollForPodWithImage(ctx, build.Image, podPollTimeoutSynclet)
+	pID, err := sbd.kCli.PollForPodWithImage(ctx, image, podPollTimeoutSynclet)
+	if err != nil {
+		return DeployInfo{}, errors.Wrapf(err, "PodWithImage (img = %s)", image)
+	}
+
+	nodeID, err := getNodeID(ctx, sbd.kCli, image)
 	if err != nil {
-		return "", fmt.Errorf("PodWithImage (img = %s): %v", build.Image, err)
+		return DeployInfo{}, errors.Wrapf(err, "error getting nodeID for image '%s'", image)
+	}
+
+	// note: this is here both to get sCli for the call to getContainerForBuild below
+	// *and* to preemptively set up the tunnel + client
+	// (i.e., we'd still want to call this to set up the client even if we were throwing away
+	// sCli)
+	sCli, err := sbd.syncletClientManager.ClientForNode(ctx, nodeID)
+	if err != nil {
+		return DeployInfo{}, errors.Wrapf(err, "error getting synclet client for node '%s'", nodeID)
 	}
 
 	// get container that's running the app for the pod we found
-	cID, err := sbd.sCli.GetContainerIdForPod(ctx, pID)
+	cID, err := sCli.GetContainerIdForPod(ctx, pID)
+	if err != nil {
+		return DeployInfo{}, errors.Wrapf(err, "syncletClient.GetContainerIdForPod (pod = %s)", pID)
+	}
+
+	logger.Get(ctx).Verbosef("talking to synclet client for node %s", nodeID.String())
+
+	return DeployInfo{cID, nodeID}, nil
+}
+
+func getNodeID(ctx context.Context, kCli k8s.Client, image reference.NamedTagged) (k8s.NodeID, error) {
```

Comment:
wait, we already have the pod ID at the point where this is called...?

Reply:
whoops, good catch!